### PR TITLE
Refactoring event based projection unit test generation template

### PR DESCRIPTION
**Pull request by hamzajg**

### CODE REVIEW

**Review by VaughnVernon — APPROVED** [review]
@hamzajg 👍 Let's also have @danilo-ambrosio review and merge.

**Comment by danilo-ambrosio** [comment]
> @hamzajg 👍 Let's also have @danilo-ambrosio review and merge.

Will review it today @hamzajg . Thanks!

File: src/main/resources/codegen/java/EntityStateProjectionUnitTest.ftl [hunk 1]
```diff
@@ -19,38 +29,209 @@ import static org.junit.jupiter.api.Assertions.assertEquals;
...
-    world = World.startWithDefaults("test-state-store-projection");;
-    new StateAdapterProvider(world).registerAdapter(${aggregateProtocolName}State.class, new ${aggregateProtocolName}StateAdapter());
-    stateStore = world.actorFor(StateStore.class, InMemoryStateStoreActor.class, Collections.singletonList(dispatcher));
-    new StatefulTypeRegistry(world).register(new StatefulTypeRegistry.Info(stateStore, ${aggregateProtocolName}State.class, ${aggregateProtocolName}State.class.getSimpleName()));
-    entity = world.actorFor(${aggregateProtocolName}.class, ${entityName}.class, "#1");
+    world = World.startWithDefaults("test-state-store-projection");
+
+    store = world.actorFor(StateStore.class, InMemoryStateStoreActor.class, Arrays.asList(new NoOpDispatcher()));
+
+    projection = world.actorFor(Projection.class, WarbleStateStoreProjection.class, store);
```
Comment (danilo-ambrosio) [inline src/main/resources/codegen/java/EntityStateProjectionUnitTest.ftl:43]: @hamzajg `WarbleStateStoreProjection` should be replaced by the actual name of the `ProjectionActor` class name of each Aggregate.

Comment (hamzajg) [reply src/main/resources/codegen/java/EntityStateProjectionUnitTest.ftl:43]: yes, the Entity State Operation Based Projection fixed in this PR #114

**Pull request merged**